src/wl_window.c

```diff
@@ -498,8 +532,12 @@ void _glfwPlatformSetCursor(_GLFWwindow* window, _GLFWcursor* cursor)
     if (window->cursorMode == GLFW_CURSOR_NORMAL)
     {
         if (cursor == NULL)
-        {
-            image = _glfw.wl.defaultCursor->images[0];
+            image = wl_cursor_theme_get_cursor(_glfw.wl.cursorTheme,
+                                               "left_ptr")->images[0];
```

Comment:
This can crash in the (unlikely) case where the current theme doesn’t have a `left_ptr` cursor, but better take care of this case and abort with an error.